bin/go-rpm-integration

```diff
@@ -155,45 +154,49 @@ while [ $# -gt 0 ] ; do
 done
 
 installfile() {
-  goipath="${1}"
-  file="${2}"
-  file="${file#./}"
-  [[ -d "${file}" && ! -L "${file}" ]] && srcdir="${file}" || srcdir=$(dirname "${file}")
-  destdir="${prefix}${gopath}/src/${goipath}/${srcdir}"
-  destdir="${destdir%/.}"
-  dir="${destdir}"
-  dirs=()
-  while [[ ! -e "${dir}" ]] ; do
-    dirs=("$dir" "${dirs[@]}")
-    dir=$(dirname "${dir}")
-  done
-  for dir in "${dirs[@]}" ; do
-    install -m 0755 -vd "${dir}"
-    if $(echo "${dir}" | grep -q "^${prefix}${gopath}/src/${goipath}") ; then
-      touch -r             ".${dir#${prefix}${gopath}/src/${goipath}}" "${dir}"
-    fi
-    echo "%dir \"${dir#${prefix}}\"" >> ${filelist}
-  done
-  if [[ -L "$file" ]] ; then
-    ln -s $(readlink "${file}") "${destdir}/$(basename ${file})"
-    touch -h      -r "${file}"  "${destdir}/$(basename ${file})"
+local goipath="${1}"
+local file="${2}"
+for proot in "$(realpath -sm ${PWD})" \
+             "${gopath}/src/${goipath}" \
+             "${prefix}${gopath}/src/${goipath}"; do
+  [[ ${file}/ == ${proot}/* ]] && file=".${file#${proot}}"
+done
+file="$(realpath -sm ${file})"
+local workdir="$(realpath -sm .)"
+if [[ ${file}/ == ${workdir}/* ]] ; then
```

Comment:
Why do I need to know the current working directory if all the code is supposed to live under `${prefix}${gopath}/src/${gopath}` anyway?

Reply:
People can path goinstall absolute directory names to create and not everything that will be created exists in the project archive (typically, anything between %{gopath}/src and the level where .goipath will exist)

The routine needs to distiguish between stuff that exists in the original archive, and that needs timestamp preservation, and elements created from scratch